packages/stream-metadata/src/contract-utils.ts

```diff
@@ -0,0 +1,14 @@
+import { ethers } from 'ethers'
+import { SpaceDapp } from '@river-build/web3'
+
+import { config } from './environment'
+
+let spaceDapp: SpaceDapp | undefined
+
+export function getSpaceDapp(): SpaceDapp {
+	if (!spaceDapp) {
+		const provider = new ethers.providers.StaticJsonRpcProvider(config.baseChainRpcUrl)
+		spaceDapp = new SpaceDapp(config.web3Config.base, provider)
+	}
+	return spaceDapp
+}
```

Comment:
@tak-hntlabs what's the reason for the indirection? why not

```
export const spaceDapp = new SpaceDapp(config.web3Config.base, provider);
```

Reply:
i was thinking of letting this indirection handle baseChainRpc connection failure due to temporary network issue. Subsequent calls can get a new spaceDapp instance once the network recovers. But I don't have enough code here to handle the network issues properly. I'll simplify it for now.